### ISSUE TITLE
gzopen64

### ISSUE THREAD

**Issue by wwhsiao** [pysam-developers/pysam#235]
Hi,

I installed the latest version of pysam (0.8.4) by issuing sudo pip install pysam. The installation seemed to have gone well.  However, when I try to use checkM ( data update command) which has pysam dependency, I got the following error: 

>sudo /usr/local/bin/checkm data update
Traceback (most recent call last):
  File "/usr/local/bin/checkm", line 36, in <module>
    from checkm import main
  File "/usr/local/lib/python2.7/site-packages/checkm/main.py", line 28, in <module>
    from checkm.resultsParser import ResultsParser
  File "/usr/local/lib/python2.7/site-packages/checkm/resultsParser.py", line 34, in <module>
    from checkm.coverage import Coverage
  File "/usr/local/lib/python2.7/site-packages/checkm/coverage.py", line 30, in <module>
    import pysam
  File "/usr/local/lib/python2.7/site-packages/pysam/__init__.py", line 1, in <module>
    from pysam.libchtslib import *
ImportError: /usr/local/lib/python2.7/site-packages/pysam/libchtslib.so: undefined symbol: gzopen64

I'm running RHEL 6.2 which has older version of zlib (1.2.3).  Is there a way for me to specify a different zlib library during pysam installation to avoid the gzopen64 error?

Thanks.

**Comment by jmarshall** [comment]
See also [samtools-help "gzopen64 error"](https://sourceforge.net/p/samtools/mailman/message/34872105/) and the surrounding thread.

**Comment by AndreasHeger** [comment]
Thanks, @jmarshall. 

@wwhsiao, has this solved your problem? Note that you can set compiler and linker flags by:

```
CFLAGS="-I/usr/local/otherdir/calg/include"  \
LDFLAGS="-L/usr/local/otherdir/calg/lib"     \
    python setup.py install
```

**Comment by wwhsiao** [comment]
Thanks Andreas. I was able to resolve the zlib conflict after following @jmarshall's advices. Also, I was able to install via pip (v. 0.8.4) using the built-in htslib. I have installed htslib 1.3 on /usr/local/lib and tried to use that library by specifying
export HTSLIB_LIBRARY_DIR=/usr/local/lib
export HTSLIB_INCLUDE_DIR=/usr/local/include
before running pip but that didn't work for me as pysam couldn't find the libhts.so after.  I wonder if that's an incompatibility between pysam 0.8.4 and htslib 1.3.  Would pysam 0.8.5 be push out via pip soon?

In any case, I think this issue can be closed and thank you @jmarshall and @AndreasHeger for helping.
